File: app/src/main/java/com/bas/google_book_app/network_service/RetrofitServiceController.java
package com.bas.google_book_app.network_service;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

import static com.bas.google_book_app.utilsdata.Constants.BOOK_BASE_URL;

public class RetrofitServiceController {

    private static Retrofit sRetrofit = null;

    public static Retrofit getClient() {
        if (sRetrofit == null) {
            sRetrofit = new Retrofit.Builder()
                    .baseUrl(BOOK_BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }
        return sRetrofit;
    }
}

File: app/src/main/java/com/bas/google_book_app/network_service/RetrofitServiceController.kt
package com.bas.google_book_app.network_service

import com.bas.google_book_app.utilsdata.Constants
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object RetrofitServiceController {
    private var sRetrofit: Retrofit? = null
    fun getClient(): Retrofit? {
        if (sRetrofit == null) {
            sRetrofit = Retrofit.Builder()
                .baseUrl(Constants.BOOK_BASE_URL)
                .addConverterFactory(GsonConverterFactory.create())
                .build()
        }
        return sRetrofit
    }
}